rllib/env/utils/infinite_lookback_buffer.py

```diff
@@ -533,9 +533,18 @@ def _get_int_index(
     ):
         data_to_use = self.data
         if _ignore_last_ts:
-            data_to_use = self.data[:-1]
+            if self.finalized:
+                data_to_use = tree.map_structure(lambda s: s[:-1], self.data)
+            else:
+                data_to_use = self.data[:-1]
         if _add_last_ts_value is not None:
-            data_to_use = np.append(data_to_use.copy(), _add_last_ts_value)
+            if self.finalized:
```

Comment:
Yeah, I remember this location to be part of the error output.

Reply:
Perfect! Thanks for confirming! This was a pretty common bug you would run into when your multi-agent episode is a little bit more unstructured wrt agents stepping patterns.